File: 1863-sum-of-all-subset-xor-totals/1863-sum-of-all-subset-xor-totals.py
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        
        xor_elements = lambda v: reduce(lambda a,b: a ^ b, v) if(v) else 0
        
        def subsets(nums, op = []):
            if(not nums): return xor_elements(op)
            
            return subsets(nums[1:], op + [nums[0]])\
        + subsets(nums[1:], op)
        
        return subsets(nums)
            

File: 1863-sum-of-all-subset-xor-totals/1863-sum-of-all-subset-xor-totals.go
func subsetXORSum(nums []int) int {
    var res = 0
    var genSubsets func(index, val int)
    genSubsets = func(index, val int) {
        if index >= len(nums) { 
            res += val
            return 
        }
        genSubsets(index + 1, val)
        genSubsets(index + 1, val ^ nums[index])
    }
    
    genSubsets(0, 0)
    return res
}

